src/simulated_annealing_kp_problem.py

```diff
@@ -0,0 +1,74 @@
+"""
+    Simulated annealing for the knapsack problem.
+"""
+import math
+import random
+
+
+# pylint: disable=R0913
+# pylint: disable=too-many-arguments
+
+class MySAProblem:
+    """
+        Class for the functions that solve the knapsack problem with SA.
+    """
+
+    def __init__(self, ksp):
+        """
+            Initialize parameters.
+        """
+        self.ksp = ksp
+        self.state = self.generate_initial_state()
+
+    def generate_initial_state(self):
+        """
+            Randomly generating the initial state of the elements in the knapsack.
+        """
+        state = []
+        for _ in range(0, len(self.ksp.data)):
+            state.append(random.randint(0, 1))
+        return state
+
+    def randomly_change_state(self):
+        """
+            Randomly generating the initial state of the elements in the knapsack.
+        """
+        changed_state = self.state
```

Comment:
- You called the function with one parameter: `neighbor = self.randomly_change_state(current_state)`, so I think you forget to update your own function because it has only the self as a parameter!
- Write `changed_state = state.copy()` instead of `changed_state = self.state`, because `neighbor = self.randomly_change_state(current_state)` will modify the current_state and here: `cost_val_diff = self.ksp.calculate_value(current_state) - self.ksp.calculate_value(neighbor)` the current_state and neighbor will be equal with each other.